equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java

```diff
@@ -52,8 +57,11 @@ private static Class<? extends CrmFormModelInterface> getCrmFormTypeMapClass(Int
 
     public <T extends CrmFormModelInterface> T getCrmFormData(CrmFormDetailsCriteriaDTO crmFormDetailsCriteriaDTO) {
         JSONObject crmFileJsonObject = getCrmFileJson(crmFormDetailsCriteriaDTO);
-
         // Format sanity checks and conversions
+        String formType = (String)crmFileJsonObject.get(CRM_FORM_FAMILY_PATH);
```

Comment:
We could use the type parameter already in the criteriaDTO instead. So, no need for additional xml/json read or parse and also remove some redundant constants

Reply:
in this case we need to modify the json response before sending to mapper

Reply:
that's not what I meant. I mean you do not need the variable formType. You can do:
```
if (crmFormDetailsCriteriaDTO.type() == CRM_TYPE_14)
```
before calling the function. Or jus have that check inside `convertCrmFormChargesBrought` but pass as an input parameter